nshlib/nsh_fscmds.c

```diff
@@ -1675,8 +1687,26 @@ int cmd_rmdir(FAR struct nsh_vtbl_s *vtbl, int argc, char **argv)
  ****************************************************************************/
 
 #if CONFIG_NFILE_STREAMS > 0 && !defined(CONFIG_NSH_DISABLESCRIPT)
-#ifndef CONFIG_NSH_DISABLE_SH
+#if !defined(CONFIG_NSH_DISABLE_SH) && defined(CONFIG_SYSTEM_SYSTEM)
 int cmd_sh(FAR struct nsh_vtbl_s *vtbl, int argc, char **argv)
+{
+  char cmd[CONFIG_NSH_LINELEN];
+
+  strncpy(cmd, "source ", sizeof(cmd) - 1);
+  strncat(cmd, argv[1], sizeof(cmd) - 1);
+
+  return system(cmd);
```

Comment:
There is a logic flaw here.  The system() call will block until the cmd is executed.  Any & or redirection on the command will not be handled.  That is incorrect and must be fixed.

Reply:
Or does the system command run on a pthread.  In that case, then the & background option would work (although it ties up both a pthread and a task).

But the redirection as it is implemented for cmd_sh() will not work for system(cmd).  Within NSH, redirection is handled via nsh_output(), stdin/stdout/stderr are not changed.

So even though you redirect the output at cmd_sh, will not be redirected by system().  This is not good.  This is a good idea but not ready to come into the master.